### PR TITLE
Track whether GitHub App is installed on org

### PR DESCRIPTION
**Pull request by eleftherias**
# Summary

In preparation for https://github.com/stacklok/epics/issues/280, we need to know whether an app was installed on an organization or a personal user account, to know which API to call.
This adds an `isOrg` column to the installations DB table, similar to what we have for access tokens.

Note: I don't think keeping track of the owner / org in the credentials is the right place. Eventually it could move to a GitHub metadata table that is specific to GitHub providers. This warrants further discussion and a design document. For now, I've followed the same pattern we're already using for access tokens.

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [x] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

**Comment by JAORMX** [comment]
I agree it should be a provider setting and not this, but oh well.

File: database/migrations/000048_installation_metadata.up.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+BEGIN;
+
+ALTER TABLE provider_github_app_installations RENAME COLUMN organization_id TO owner_id;
```
[the remark this answers is not in the document]
Comment (eleftherias) [reply database/migrations/000048_installation_metadata.up.sql:17]: Thanks for pointing that out. I can remove the rename, it's not necessary.

**Comment by eleftherias** [comment]
> I seem to recall that some (many?) of the GitHub operations had endpoints that worked for both users and orgs. I'm assuming that packages aren't one of those endpoints, given the other limitations.

That's right. Packages only have per-user or per-org endpoints.

File: internal/providers/service.go [hunk 2]
```diff
@@ -58,6 +58,9 @@ type ProviderService interface {
 	DeleteProvider(ctx context.Context, provider *db.Provider) error
 }
 
+// TypeGitHubOrganization is the type returned from the GitHub API when the owner is an organization
+const TypeGitHubOrganization = "Organization"
```
Comment (jhrozek) [inline internal/providers/service.go:62]: Nit: doesn't have to be exported
(not needed in this PR even if you agree the const doesn't have to be exported)

**Review by jhrozek — COMMENTED** [review]
The code makes sense to me and I think @evankanderson 's suggestions have been addressed. I think this PR is ready to be approved once rebased.

**Review by jhrozek — APPROVED** [review]
I think the code is good now and all the tests failures are a result of github having a bad day.

**Pull request merged**